chew.go

```diff
@@ -96,6 +263,19 @@ Process takes a list of URLs and returns a list of Chunks
 The slice of strings to be processed can be URLs or file paths
 The context is optional and can be used to cancel the processing
 of the URLs after a certain amount of time
+
+This function is safe for concurrent use.
+
+Usage:
+
+	chunks, err := chew.Process([]string{"https://example.com", "file://path/to/file.txt"})
+	if err != nil {
+		log.Fatalf("Error processing URLs: %v", err)
+	}
+
+	for _, chunk := range chunks {
+		log.Printf("Chunk: %s\n Source: %s\n", chunk.Content, chunk.Source)
+	}
 */
 func Process(urls []string, ctxs ...context.Context) ([]common.Chunk, error) {
```

Comment:
it's too later but ctx should always be the first argument, here you used a variadic trick to solve the issues, but it's a bit dirty, especially if you consider the variadic can be used to pass more than one context, and your code will silently only use the first one

If it's not too late, I would change the signature

Reply:
The number of your stars, and the fact there is no public project using chew for now, means it's not too late

https://pkg.go.dev/github.com/mmatongo/chew
> Imported by: 0